### PR TITLE
Remove observer_stdout/observer_stderr from Observer trait

### PR DESCRIPTION
**Pull request by tokatoka**
and change how command executor sets stuff up.

### CODE REVIEW

File: fuzzers/nyx_libxml2_parallel/src/main.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ fn main() {
         let mut feedback = MaxMapFeedback::new(&observer);
         let mut objective = CrashFeedback::new();
         let scheduler = RandScheduler::new();
-        let mut executor = NyxExecutor::new(helper, tuple_list!(observer));
+        let mut executor = NyxExecutorBuilder::new().build(helper, tuple_list!(observer));
```
Comment (domenukk) [inline fuzzers/nyx_libxml2_parallel/src/main.rs:57]: What's our normal naming scheme btw? Should this be `::builder` instead of `new` (or something else?)

Comment (tokatoka) [reply fuzzers/nyx_libxml2_parallel/src/main.rs:57]: builder is better i'll change

File: libafl/src/executors/command.rs [hunk 2]
```diff
@@ -189,134 +173,36 @@ impl CommandExecutor<(), (), ()> {
...
-            phantom: PhantomData,
-        })
-    }
-
-    /// Parses an AFL-like commandline, replacing `@@` with the input file
-    /// generated by the fuzzer (similar to the `afl-fuzz` command line).
-    ///
-    /// If no `@@` was found, will use stdin for input.
-    /// The arg 0 is the program.
-    pub fn parse_afl_cmdline<IT, O>(
```
[the remark this answers is not in the document]
Comment (domenukk) [reply libafl/src/executors/command.rs:271]: the builder should then probably have a parse_afl_cmdline option, right?

File: fuzzers/nyx_libxml2_parallel/src/main.rs (hunk 1 above)
Comment (domenukk) [reply fuzzers/nyx_libxml2_parallel/src/main.rs:57]: Ok

File: libafl/src/executors/command.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tokatoka) [reply libafl/src/executors/command.rs:271]: i want to remove it because that specific "@@" parsing is specific to forkserver and this is not

**Pull request merged**